src/impl_/trampoline.rs

```diff
@@ -178,10 +177,9 @@ where
     R: PyCallbackOutput,
 {
     let trap = PanicTrap::new("uncaught panic at ffi boundary");
-    // Necessary to construct a pool until PyO3 0.22 when the GIL Refs API is fully disabled
-    #[allow(deprecated)]
-    let pool = unsafe { GILPool::new() };
-    let py = pool.python();
+
+    let guard = GILGuard::acquire();
```

Comment:
I think here it would be nice to have a `GILGuard::assume()` or similar to avoid the need to ever call `PyGILState_Ensure` here (although we do still want to increment `GIL_COUNT`).

Reply:
Hmm, will `gil_is_acquired` not return true in all cases? (Or not do so with sufficient speed?)

Reply:
Done.